src/background_music/LoudnessFrameProcessor.h

```diff
@@ -0,0 +1,189 @@
+#ifndef AUDACIOUS_PLUGINS_BGM_LOUDNESS_FRAME_PROCESSOR_H
+#define AUDACIOUS_PLUGINS_BGM_LOUDNESS_FRAME_PROCESSOR_H
+/*
+ * Background music (equal loudness) Plugin for Audacious
+ * Copyright 2023 Michel Fleur
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions, and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the following disclaimer in the documentation
+ *    provided with the distribution.
+ *
+ * This software is provided "as is" and without any warranty, express or
+ * implied. In no event shall the authors be liable for any damages arising from
+ * the use of this software.
+ */
+#include "Integrator.h"
+#include "Loudness.h"
+#include "basic_config.h"
+#include <cmath>
+#include <libaudcore/runtime.h>
+
+class LoudnessFrameProcessor
+{
+    static constexpr float SHORT_INTEGRATION = 0.4;
+    static constexpr float LONG_INTEGRATION = 6.3;
+    /*
+     * This adjusts the slow RMS measurement so that it's displayed correctly
+     * in the audacious VU meter. This only helps for expectation management.
+     */
+    static constexpr float SLOW_VU_FUDGE_FACTOR = 2.0f;
+    static constexpr float FAST_VU_FUDGE_FACTOR = 3.0f;
+
+    FastAttackSmoothRelease release_integration;
+    Integrator long_integration;
+    PerceptiveRMS perceivedLoudness;
+    float slow_weight = 0;
+    float target_level = 0.1;
+    float maximum_amplification = 1;
+    float perception_slow_balance = 0.3;
+    float minimum_detection = 1e-6;
+    RingBuf<float> read_ahead_buffer;
+    int channels_ = 0;
+    int processed_frames = 0;
+
+    static float get_clamped_value(const char * variable, const double minimum,
+                                   const double maximum)
+    {
+        return static_cast<float>(std::clamp(
+            aud_get_double(CONFIG_SECTION_BACKGROUND_MUSIC, variable),
+            static_cast<double>(minimum), static_cast<double>(maximum)));
```

Comment:
`static_cast<double>` is redundant since `minimum` and `maximum` are already `double`

Reply:
Indeed! I did switch most internal calculations to float, so it was probably a leftover from that.
I also added the header <algorithm> in Integrator.h that defines std::max(a,b)